datmo/cli/command/task.py

```diff
@@ -47,15 +48,14 @@ def run(self, **kwargs):
         self.cli_helper.echo(__("info", "cli.task.run"))
 
         # Create input dictionaries
-        snapshot_dict = {
-            "environment_definition_filepath":
+        snapshot_dict = {}
+        if kwargs['environment_definition_filepath']:
+            snapshot_dict["environment_definition_filepath"] =\
                 kwargs['environment_definition_filepath']
-        }
-
         if not isinstance(kwargs['cmd'], list):
             if platform.system() == "Windows":
                 kwargs['cmd'] = kwargs['cmd']
-            else:
+            elif type(kwargs['cmd']) is str:
                 kwargs['cmd'] = shlex.split(kwargs['cmd'])
```

Comment:
So if it's not a list, we simply pass the thing in windows. Why?
For all other systems, we check if it's a string and then split it.      What about all other types?   I think we need aN else:  clause and raise an exception

Reply:
this is what i looked at: https://stackoverflow.com/questions/33560364/python-windows-parsing-command-lines-with-shlex

don't really understand why it works but i suppose its because of the explanation given in the first answer